### PR TITLE
add predicates to match based on jwt content

### PR DESCRIPTION
**Pull request by szuecs**
JwtAnyKV("iss", "https://accounts.google.com")
JwtAllKV("iss", "https://accounts.google.com", "email", "skipper-router@googlegroups.com")

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

**Comment by mikkeloscar** [comment]
@szuecs regarding the naming, would it make sense to include `Payload` in the name, because this is what it looks into afaict.? I use the term payload because this is what's used here: https://jwt.io/

**Comment by njuettner** [comment]
@szuecs JWT should be written in uppercase WDYT?

**Comment by szuecs** [comment]
@mikkeloscar @njuettner addressed your comments

**Comment by szuecs** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

File: predicates/auth/jwt.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
+/*
+Package auth implements custom predicates to match based on content
+of the HTTP Authorization header.
+
+You can use this predicate to match a route based on data in the 2nd part of a JWT token, for example based on the issuer.
```
Comment (aryszka) [inline predicates/auth/jwt.go]: I think that it is nicer to use the passive form of "This predicate can be used..." in code docs.

File: predicates/auth/jwt.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+	"github.com/zalando/skipper/routing"
+)
+
+const (
+	authHeaderName   = "Authorization"
+	authHeaderPrefix = "Bearer "
+)
+
+var InvalidArgsError = errors.New("invalid arguments")
```
Comment (aryszka) [inline predicates/auth/jwt.go]: errors should be called conventionally ErrSomething. -> ErrInvalidArgsError. For some reason we also have this error here: https://github.com/zalando/skipper/blob/master/predicates/predicates.go#L6

**Comment by aryszka** [comment]
:+1:

**Comment by aryszka** [comment]
formatting fails:

```
make[1]: *** [check-fmt] Error 1
```

**Comment by aryszka** [comment]
:+1:

**Pull request merged**